### PR TITLE
Decrypt attachments

### PR DESCRIPTION
**Pull request by KamilaHasanbega**
- Fixed decrypting attachments
- Fixed the "should show image attachments as other attachments"
- Fixed the case where inputting attachments of type .txt would be shown as normal email-content text

### CODE REVIEW

File: chrome/components/gmail/gmail-service.js [hunk 1]
```diff
@@ -1639,12 +1638,12 @@ GmailService.prototype.sendRFC2822Message_ = function(threadId, content) {
  * @param {string} content The email content.
  * @param {?string} subject The email subject.
  * @param {Array<string>} recipients An array of email addresses.
- * @param {Array<Object>=} opt_attachments Attachment files.
+ * @param {Array<Object>=} opt_attachments
```
Comment (koto) [inline chrome/components/gmail/gmail-service.js]: Missing parameter description.

File: chrome/components/gmail/gmail-service.js [hunk 2]
```diff
@@ -1889,19 +1887,30 @@ GmailService.prototype.mimeTreeWalker_ = function(rootNode) {
   var ct = ctHeader.value.toLowerCase();
   var enc = encHeader.value.toLowerCase();
 
-  // Case 1: Single plaintext node.
+  // Case 1: Attachment
+
+  if (goog.isDefAndNotNull(rootNode.header[
```
Comment (koto) [inline chrome/components/gmail/gmail-service.js]: Some of those checks can make it into a single line. It's hard to read that conditions. You might extract some of them to a Boolean variable (e.g. var hasContentDisposition = (... == ... && ... ); or to a function.

File: chrome/components/gmail/gmail-service.js [hunk 3]
```diff
@@ -2016,7 +1997,8 @@ GmailService.prototype.isMime_ = function(message) {
  * @return {boolean} Returns true if the string is valid base64, otherwise false
  * @private
  */
-GmailService.prototype.isValidBase64_ = function(encodedString) {
+GmailService.prototype.isValidBase64_ = function(
+    encodedString) {
```
Comment (koto) [inline chrome/components/gmail/gmail-service.js]: This will fit into a previous line.

File: chrome/components/gmail/gmail-service.js [hunk 4]
```diff
@@ -2029,12 +2011,28 @@ GmailService.prototype.isValidBase64_ = function(encodedString) {
  * @return {{content: string, type: string}}
  * @private
  */
-GmailService.prototype.prepareContentForDisplay_ = function(content, type) {
+GmailService.prototype.prepareContentForDisplay_ = function(
+    content, type) {
```
Comment (koto) [inline chrome/components/gmail/gmail-service.js]: Same here.

File: chrome/components/gmail/gmail-service.js [hunk 5]
```diff
@@ -2029,12 +2011,28 @@ GmailService.prototype.isValidBase64_ = function(encodedString) {
...
   return {content: content, type: type};
 };
 
 
 /**
+ * Inserts attachment into an object that can be
```
Comment (koto) [inline chrome/components/gmail/gmail-service.js]: This description would fit into a single line.

File: chrome/components/gmail/gmail-service.js [hunk 6]
```diff
@@ -2029,12 +2011,28 @@ GmailService.prototype.isValidBase64_ = function(encodedString) {
...
+ * displayed within the E2EMail app.
+ * @param {string} url The blob url
+ * @param {string} type The type of the attachment
+ * @param {string} name The name of the attachment
+ * @return {{url: string, type: string, filename: string}}
+ * @private
+ */
+GmailService.prototype.prepareAttachmentForDisplay_ = function(
+    url, type, name) {
```
Comment (koto) [inline chrome/components/gmail/gmail-service.js]: We only move parameters in f. declaration to a separate line if they don't fit the 80 char limit.

**Comment by koto** [comment]
Some tests are failing - some of them related to the (now removed) image support, and something about utf-8 in base64 encoding.

File: chrome/components/gmail/gmail-service_test.js [hunk 7]
```diff
@@ -823,3 +813,4 @@ describe('GmailService', function() {
 });
 
 });  // goog.scope
+git
```
Comment (koto) [inline chrome/components/gmail/gmail-service_test.js]: ? This line should be removed.

**Comment by koto** [comment]
LGTM, safe to merge. Fixes #25 and #30.

**Comment by kbsriram** [comment]
rebased via 0ce7c67

**Pull request closed without merging**